src/net.h

```diff
@@ -782,6 +789,80 @@ class CNode
     std::atomic_bool fPauseRecv{false};
     std::atomic_bool fPauseSend{false};
 
+    bool IsOutboundOrBlockRelayConn() const {
+        switch(m_conn_type) {
+            case ConnectionType::OUTBOUND:
+            case ConnectionType::BLOCK_RELAY:
+                return true;
+            case ConnectionType::INBOUND:
+            case ConnectionType::MANUAL:
+            case ConnectionType::ADDR_FETCH:
+            case ConnectionType::FEELER:
+                return false;
+        }
+
+        assert(false);
+    }
+
+    bool IsFullOutboundConn() const {
+        return m_conn_type == ConnectionType::OUTBOUND;
+    }
+
+    bool IsManualConn() const {
+        return m_conn_type == ConnectionType::MANUAL;
+    }
+
+    bool IsBlockOnlyConn() const {
+        return m_conn_type == ConnectionType::BLOCK_RELAY;
+    }
+
+    bool IsFeelerConn() const {
+        return m_conn_type == ConnectionType::FEELER;
+    }
+
+    bool IsAddrFetchConn() const {
+        return m_conn_type == ConnectionType::ADDR_FETCH;
+    }
+
+    bool IsInboundConn() const {
+        return m_conn_type == ConnectionType::INBOUND;
+    }
+
+    bool IsUniqueNetBlockConn() const {
```

Comment:
I don't love this (or `ExpectServicesFromConn()` below) where logic that is only used in one calling function is placed inside the class. `IsUniqueNetBlockConn()` is very specific to `ThreadOpenConnections()` so it makes sense to me that the logic and comment should live in that function.

I realise this sounds like a contradiction of https://github.com/bitcoin/bitcoin/pull/19316#issuecomment-655057621, but to me these very specific cases don't belong inside the object.

Reply:
I don't have a strong preference either way, but agree with you around `IsUniqueNetBlockConn` and `ExpectServicesFromConn` not being particularly helpful. Since both are implemented as switch statements, I think I'll revert to having them integrated into their calling functions. 

some of the other `m_conn_type` wrappers are also only used in one location. eg. `IsFeelerConn` & `IsBlockOnlyConn`. But I'm going to leave them as is. 

reviewers- if you disagree, please help me understand why!

Reply:
I ended up inlining `IsUniqueNetBlockConn`, but left `ExpectServicesFromConn` because I couldn't figure out a clear way to inline while maintaining a switch statement.